src/System.Net.Http/src/uap/System/Net/HttpClientHandler.cs

```diff
@@ -486,28 +486,38 @@ private void ApplyDecompressionSetting(HttpRequestMessage request, Decompression
             {
                 if (_clientCertificates != null && _clientCertificates.Count > 0)
                 {
-                    RTCertificate cert = await CertificateHelper.ConvertDotNetClientCertToWinRtClientCertAsync(_clientCertificates[0]);
-                    if (cert == null)
+                    X509Certificate2 clientCert = CertificateHelper.GetEligibleClientCertificate(_clientCertificates);
+                    if (clientCert == null)
+                    {
+                        return;
+                    }
+
+                    RTCertificate rtClientCert = await CertificateHelper.ConvertDotNetClientCertToWinRtClientCertAsync(clientCert);
+                    if (rtClientCert == null)
                     {
                         throw new PlatformNotSupportedException(string.Format(CultureInfo.InvariantCulture,
                             SR.net_http_feature_UWPClientCertSupportRequiresCertInPersonalCertificateStore));
                     }
 
-                    _rtFilter.ClientCertificate = cert;
+                    _rtFilter.ClientCertificate = rtClientCert;
                 }
 
                 return;
```

Comment:
Nit: This "return" is no longer needed, now that the **ClientCertificateOption.Automatic** case is inside an _else_